hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -703,7 +703,14 @@ public boolean mkdirs(Path f, FsPermission permission) throws IOException {
 
   @Override
   public long getDefaultBlockSize() {
-    return (long) getConf().getStorageSize(
+    Configuration conf = getConf();
+    ConfigurationSource source;
+    if (conf instanceof OzoneConfiguration) {
+      source = (ConfigurationSource) conf;
+    } else {
+      source = new LegacyHadoopConfigurationSource(conf);
+    }
```

Comment:
Instead of duplicating this logic from `initialize`:

https://github.com/apache/ozone/blob/09579756b0756fcaf1b22b860c1ee1eb927e82c2/hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java#L161-L166

I think it would be cleaner to store this `ConfigurationSource`

* either in `BasicOzoneFileSystem`
* or in `BasicOzoneClientAdapterImpl` (and expose it via accessor method)

and then use it here.